### PR TITLE
pb-3928: Fixed issue in cleaning up the vs and vsc in the case of kdmp localsnapshot restore.

### PR DESCRIPTION
**Pull request by siva-portworx**
**What this PR does / why we need it**:
```
    pb-3928: Fixed issue in cleaning up the vs and vsc in the case of kdmp
    localsnapshot restore.

            - Now getting the restore PVC spec to get the volumesnapshot
              name and calling snapshotDriver.DeleteSnapshot api to delete vs and vsc.
```
**Which issue(s) this PR fixes** (optional)
Closes # pb-3928

**Special notes for your reviewer**:
**Testing:**
Tested kdmp localsnapshot and verified that vs and vsc are getting deleted.

### CODE REVIEW

File: pkg/controllers/dataexport/reconcile.go [hunk 1]
```diff
@@ -1481,18 +1481,21 @@ func (c *Controller) cleanupLocalRestoredSnapshotResources(de *kdmpapi.DataExpor
...
-			msg := fmt.Sprintf("failed in removing local volume snapshot CRs for %s/%s: %v", de.Status.VolumeSnapshot, de.Status.SnapshotPVCName, err)
-			if err != nil {
-				logrus.Errorf(msg)
-				return nil, false, fmt.Errorf(msg)
-			}
-		}
-
+		// Get the Restore pvc spec.
+                rpvc, err := core.Instance().GetPersistentVolumeClaim(de.Status.RestorePVC.Name, de.Namespace)
+                logrus.Infof("sivakumer getting rpvc err %v", err)
```
Comment (prashanthpx) [inline pkg/controllers/dataexport/reconcile.go:1486]: Remove log line

**Review by prashanthpx — APPROVED** [review]
LGTM

**Comment by diptiranjanpx** [comment]
deleting the pvc if ignorePVC is set to false. Are you planning to put that in a different patch?

File: pkg/controllers/dataexport/reconcile.go (hunk 1 above)
Comment (siva-portworx) [reply pkg/controllers/dataexport/reconcile.go]: Done

**Review by diptiranjanpx — APPROVED** [review]

**Comment by siva-portworx** [comment]
> deleting the pvc if ignorePVC is set to false. Are you planning to put that in a different patch?

Fixed. Thanks for point it out.

**Pull request merged**